src/main/java/io/vertx/core/http/RequestOptions.java

```diff
@@ -333,7 +333,19 @@ private URL parseUrl(String surl) {
   public RequestOptions setAbsoluteURI(String absoluteURI) {
     Objects.requireNonNull(absoluteURI, "Cannot set a null absolute URI");
     URL url = parseUrl(absoluteURI);
-    Boolean ssl = false;
+    return setAbsoluteURI(url);
+  }
+
+  /**
+   * Like {@link #setAbsoluteURI(String)} but using an {@link URL} parameter.
+   *
+   * @param url the uri to use
+   * @return a reference to this, so the API can be used fluently
+   */
+  @GenIgnore
```

Comment:
Make it a permitted type, so it can be used by rx, kotlin, and so...

Reply:
Done